### PR TITLE
Fix headers and update acknowledgements

### PR DESCRIPTION
**Pull request by GlendaChong**
- Remove numbering in headers for ease of re-ordering later
- Update Acknowledgement Section
- Update headers to pascal case
- Add planned enhancements and effort section

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -8,43 +8,33 @@ title: Developer Guide
...
--   [JavaFX CSS](https://docs.oracle.com/javafx/2/api/javafx/scene/doc-files/cssref.html) - JavaFX CSS is a simple CSS parser.
--   [JavaFX FXML](https://docs.oracle.com/javafx/2/fxml_get_started/jfxpub-fxml_get_started.htm) - JavaFX FXML is an XML-based language that provides the structure for building a user interface separate from the application logic of your code.
--   [JavaFX Scene Builder](https://gluonhq.com/products/scene-builder/) - JavaFX Scene Builder is a visual layout tool that lets users quickly design JavaFX application user interfaces, without coding.
--   [JUnit](https://junit.org/junit5/) - JUnit is a unit testing framework for the Java programming language.
--   [PlantUML](https://plantuml.com/) - PlantUML is an open-source tool allowing users to create UML diagrams from a plain text language.
+* Libraries used: 
+  * [JavaFX](https://openjfx.io/) 
+  *  [Jackson](https://github.com/FasterXML/jackson) 
+  *  [JUnit](https://junit.org/junit5/) 
+  *  [Codecov](https://codecov.io/)
```
Comment (tayruxin) [inline docs/DeveloperGuide.md:19]: wait but all of that is OG AB3 tho should we really be deleting all that libraries

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -255,13 +245,13 @@ shown in the activity diagram below.
         object. There is a need to create additional methods to update the `CompanyDetailPanel` when there is any changes
         made to the `Company` object.
 
-### 4.3 Find command
+### Find Feature
 
-#### 4.3.1 Implementation
+#### Implementation
```
Comment (tayruxin) [inline docs/DeveloperGuide.md:250]: wait are we all removing all these headers, cut right now like this it will appear in the TOC?

Comment (GlendaChong) [reply docs/DeveloperGuide.md:250]: it originally already appears in the TOC

Comment (tayruxin) [reply docs/DeveloperGuide.md:250]: Yeap, we planning to remove from the TOC? I mean either way is fine by me, but if we intend to remove from TOC then these needs to change

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (GlendaChong) [reply docs/DeveloperGuide.md:19]: I think we don't have to show which ones did AB3 used

**Review by tayruxin — APPROVED** [review]

**Pull request merged**